### PR TITLE
feat: add object detection module (YOLO, TorchGeo, SAM)

### PR DESCRIPTION
**Pull request by mmann1123**
## Summary

Adds a new `geowombat.detect` submodule for tiled, georeferenced object
detection on overhead imagery, mirroring the `fit / predict / fit_predict`
shape of `geowombat.ml` for classification.

- **Detectors**: `YOLODetector` (Ultralytics; axis-aligned + DOTA-v1 OBB
  weights), `TorchGeoDetector` (TorchVision Faster R-CNN / RetinaNet with
  optional TorchGeo pretrained weights), `SAMRefiner` (Segment Anything;
  refines boxes to polygon masks).
- **Geowombat-native accessors**: `src.gw.detect(detector, ...)` for tiled
  inference with cross-tile NMS and pixel→CRS box conversion;
  `src.gw.to_yolo_dataset(labels, class_col=..., out_dir=...)` for writing
  Ultralytics-layout training corpora from raster + vector labels.
- **Module-level API**: `predict`, `fit`, `fit_predict`, `build_dataset`,
  `boxes_from_polygons` (AABB ↔ OBB conversion with DOTA-v1 pairing
  guidance), `detection_accuracy` (per-class P/R/F1/AP at one or more IoU
  thresholds, returns a review-ready GeoDataFrame),
  `plot_detections` (TP/FP/FN matplotlib overlay), `export_for_review` /
  `recompute_from_review` (QGIS GeoPackage round-trip).
- **Sensor-aware band selection**: when `gw.config.update(sensor=...)` is
  active, `src.gw.detect` reads `src.band.values` and resolves RGB band
  indices automatically — no need to pass `band_indices` per call.

The old `ml/detection_*.py` files are renamed into the new `detect/`
package (`ml/detection_data.py → detect/data.py`,
`ml/detectors.py → detect/detectors.py`,
`ml/detection_metrics.py → detect/metrics.py`,
`tests/ml_detection_test.py → tests/detect_test.py`).

## Docs

- New [doc/source/object-detection.rst](doc/source/object-detection.rst):
  full walkthrough with prose, the "Recommended setup for aerial /
  satellite imagery" admonition (DOTA-v1 OBB weights + `oriented=True`),
  the AABB vs OBB explainer, and a "Digitizing polygons for high-quality
  OBB labels" subsection (tight polygons → clean `minimum_rotated_rectangle`
  output, with SAM-refinement and QGIS workflow tips).
- New [notebooks/object_detection.ipynb](notebooks/object_detection.ipynb)
  generated from [notebooks/_build_object_detection_ipynb.py](notebooks/_build_object_detection_ipynb.py).
  Walks NAIP from Microsoft Planetary Computer + OSM building footprints
  through dataset construction, pretrained vs fine-tuned inference,
  metrics comparison, QGIS review export, and SAM polygon refinement.
  Symlinked into `doc/source/object_detection.ipynb` and rendered by
  `nbsphinx` (with `nbsphinx_execute = 'never'` so cached outputs ship
  without re-execution at build time).
- `nbsphinx` + `ipykernel` added to the `[docs]` extras.

## Tests

- [tests/detect_test.py](tests/detect_test.py) — 459 lines covering the
  detection pipeline.

### CODE REVIEW

**Comment by mmann1123** [comment]
@jgrss Need a review on object detection module

File: src/geowombat/backends/rasterio_.py [hunk 1]
```diff
@@ -997,10 +997,6 @@ def warp(
                 'width': dst_window.width,
                 'nodata': None,
                 'warp_mem_limit': warp_mem_limit,
-                'warp_extras': {
```
Comment (mmann1123) [inline src/geowombat/backends/rasterio_.py:1000]: this helps avoid some GDAL warnings and improves perf

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -54,5 +54,5 @@ jobs:
         uses: actions/upload-artifact@v4
         with:
           name: documentation
-          path: doc/build/html/
+          path: doc/_build/html/
```
Comment (mmann1123) [inline .github/workflows/docs.yml:57]: I updated the docs a bit and added the notebooks directly into the read the docs page

File: src/geowombat/backends/rasterio_.py (hunk 1 above)
Comment (jgrss) [reply src/geowombat/backends/rasterio_.py:1000]: Not easy to keep as user options?

File: src/geowombat/detect/_tiling.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+object detection (training-data tiling and inference). Detection-time
+tiles must overlap so that objects on tile seams can still be matched;
+``.gw.windows()`` is non-overlapping by design, so we add the stride
+layer here rather than changing the accessor.
+"""
+
+from rasterio.windows import Window
+
+
+def overlapped_windows(src, tile_size, overlap=0.0):
```
Comment (jgrss) [inline src/geowombat/detect/_tiling.py]: minor: you could type hint the output as a `Generator`.

File: src/geowombat/detect/api.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    detector,
+    labels,
+    class_col,
+    out_dir,
+    tile_size=640,
+    overlap=0.1,
+    epochs=50,
+    predict_kwargs=None,
+    **dataset_kwargs,
+):
```
Comment (jgrss) [inline src/geowombat/detect/api.py]: General comment: could add output type hints

**Review by jgrss — APPROVED** [review]

File: src/geowombat/backends/rasterio_.py (hunk 1 above)
Comment (mmann1123) [reply src/geowombat/backends/rasterio_.py:1000]: On rasterio>=1.4 / GDAL>=3.12, any non-empty warp_extras or multi kwarg on WarpedVRT makes GDAL print Warning 6: warp options does not support option WARP_EXTRAS/MULTI once per construction — and dask reconstructs the VRT per chunk, so it spammed.

So I restored threading the clean way: rio.Env(GDAL_NUM_THREADS=str(num_threads)) wrapping the VRT construction and chunk reads, only opened when num_threads > 1. Zero warnings, real ~22% speedup on chunked reads at threads=2 (verified on a 448MB synthetic raster).

For "arbitrary user GDAL options" — users can already pass anything via rio.Env(...) around gw.open / gw.save. I verified rasterio Env nesting merges correctly, so a user's outer env composes with our inner one without overriding. No new public surface needed; just documented the pattern in io.rst.

File: src/geowombat/detect/_tiling.py (hunk 3 above)
Comment (mmann1123) [reply src/geowombat/detect/_tiling.py]: Added -> T.Generator[T.Tuple[int, int, Window], None, None] plus parameter annotations

File: src/geowombat/detect/api.py (hunk 4 above)
Comment (mmann1123) [reply src/geowombat/detect/api.py]: Added return types: predict -> gpd.GeoDataFrame, fit -> T.Any, fit_predict -> T.Tuple[gpd.GeoDataFrame, dict]
